### PR TITLE
Add educational code pointers to top level README

### PR DESCRIPTION
**Pull request by wconstab**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #457

image
looks like this, and the links work

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,6 +18,14 @@ Our guiding principles when building `torchtitan`:
 
 [![Welcome to torchtitan!](assets/images/titan_play_video.png)](https://youtu.be/ee5DOEqD35I?si=_B94PbVv0V5ZnNKE "Welcome to torchtitan!")
 
+### Dive Into the code
+
+You may want to see how the model is defined or how parallelism techniques are applied. For a guided tour, see these files first:
+* [train.py](https://github.com/pytorch/torchtitan/blob/main/train.py) - the main training loop and high-level setup code
+* [torchtitan/parallelisms/parallelize_llama.py](https://github.com/pytorch/torchtitan/blob/main/torchtitan/parallelisms/parallelize_llama.py) - helpers for applying TP/DP/PP parallelisms to the model
```
Comment (tianyu-l) [inline README.md]: nit: 
```suggestion
* [torchtitan/parallelisms/parallelize_llama.py](https://github.com/pytorch/torchtitan/blob/main/torchtitan/parallelisms/parallelize_llama.py) - helpers for applying Data / Tensor / Pipeline Parallelisms to the model
```

File: README.md [hunk 2]
```diff
@@ -18,6 +18,14 @@ Our guiding principles when building `torchtitan`:
...
+* [torchtitan/checkpoint.py](https://github.com/pytorch/torchtitan/blob/main/torchtitan/checkpoint.py) - utils for saving/loading distributed checkpoints
+* [torchtitan/models/llama/model.py](https://github.com/pytorch/torchtitan/blob/main/torchtitan/models/llama/model.py) - the LLaMa model definition (shared for llama2/llama3 variants)
```
Comment (tianyu-l) [inline README.md]:
```suggestion
* [torchtitan/models/llama/model.py](https://github.com/pytorch/torchtitan/blob/main/torchtitan/models/llama/model.py) - the Llama model definition (shared for Llama2 and Llama3 variants)
```

File: README.md [hunk 3]
```diff
@@ -18,6 +18,14 @@ Our guiding principles when building `torchtitan`:
 
 [![Welcome to torchtitan!](assets/images/titan_play_video.png)](https://youtu.be/ee5DOEqD35I?si=_B94PbVv0V5ZnNKE "Welcome to torchtitan!")
 
+### Dive Into the code
```
Comment (tianyu-l) [inline README.md]:
```suggestion
### Dive into the code
```

**Review by tianyu-l — COMMENTED** [review]
lgtm! Let's give other people a chance to take a look.

File: README.md (hunk 1 above)
Comment (awgu) [reply README.md]: https://github.com/pytorch/torchtitan/blob/07ab2f9a7298c93f3dd67b728c0c25c6c92292e2/torchtitan/parallelisms/parallelize_llama.py#L7-L8

The file also includes non-parallelism related techniques like AC and `torch.compile`. Are these worth mentioning in the readme?

**Review by awgu — APPROVED** [review]

**Pull request merged**